### ISSUE TITLE
Exceptions thrown when running under cider are not shown to the user

### ISSUE THREAD

**Issue by aaelony** [tomfaulhaber/excel-templates#18]
Thanks for a most awesome library.  

A few things I have noticed however, is that sometimes a collection of row data might fail to produce the resulting excel file and it may do so siliently, without giving a sense of the problem. One such issue that was a headscratcher involved a row like [[nil :some_keyword nil]]  that did not work until I decided to try [[nil "some_keyword" nil]] instead (which did/does work).

Any ideas on how to propagate reasons why the result file doesn't get produced and why, would be helpful.  In my case, they seem to mostly resolve by forcing sequences to be vectors (perhaps laziness is the issue?) and/or converting keywords to strings.

Thanks again for a great library

**Comment by tomfaulhaber** [comment]
Question: were you running under cider or another nrepl client?

**Comment by aaelony** [comment]
Oh, could it be cider specific?  This was under CIDER 0.9.1 (Java 1.7.0_85, Clojure 1.8.0-alpha2, nREPL 0.2.10)

What optimal emacs env/versions would you recommend?

**Comment by tomfaulhaber** [comment]
I think it might be cider-specific. The exception ends up in one of cider's buffers, though, so you can find it.

Could you do me a favor and try the same thing directly under `lein repl` and see if you see the exception there?

**Comment by aaelony** [comment]
sure, will try to get to this soon.

**Comment by aaelony** [comment]
Hi, I've confirmed that it is the case that there are errors shown via `lein repl` that are not visible in `cider`.
Feel free to close this.  Thank again. -Avram

**Comment by tomfaulhaber** [comment]
Actually, I'd like to figure out why that's happening and get the errors to show up in cider in the right way. Thanks for confirming this for me.

**Comment by aaelony** [comment]
Thank-you for thinking about this.  The peculiar error I am getting now at the repl is as follows:

**Comment by aaelony** [comment]
Thank-you for thinking about this.  The issue with keywords and strings being resolved, a peculiar error I am getting now at the repl is as follows: 

```
java.lang.IllegalArgumentException: Sheet index (-1) is out of range (0..7)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.validateSheetIndex(XSSFWorkbook.java:1061)
        at org.apache.poi.xssf.usermodel.XSSFWorkbook.setSheetName(XSSFWorkbook.java:1238)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:93)
        at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28)
        at excel_templates.build$clone_sheet_BANG_$fn__4649.invoke(build.clj:241)
        at excel_templates.build$clone_sheet_BANG_.invoke(build.clj:241)
        at excel_templates.build$create_missing_sheets_BANG_$fn__4683.invoke(build.clj:286)
        at excel_templates.build$create_missing_sheets_BANG_.invoke(build.clj:270)
        at excel_templates.build$render_to_file.invoke(build.clj:381)
```
I am not sure what exactly is causing this just yet.
Will need to pull things apart to be able to reproduce outside of my project code..

-A

**Comment by tomfaulhaber** [comment]
@aaelony Looking at that error, my guess is that you are specifying a source sheet name that doesn't exist in the template. (Of course, it could be a bug.)

**Comment by aaelony** [comment]
That was really helpful.  I should have guessed that `Sheet index (-1) is out of range (0..7)` meant that it wasn't finding the sheet name.  The specified names in the source were identical to the sheet names in the template, but 7 sheets were specified, of which 3 sheet names were very similar and a bit verbose.  Changing the names to be shorter and less similar makes it work correctly.  My guess is that there might be a call to truncate long strings somewhere?  All set, thanks again.

**Comment by tomfaulhaber** [comment]
The hidden exceptions seem to be fixed with Cider 0.11. If anyone sees this problem again, please open a new bug with details (including excel-templates version, cider version, and details to reproduce). Thanks!
